src/Microsoft.AspNet.Authentication.Cookies/CookieAppBuilderExtensions.cs

```diff
@@ -23,30 +24,7 @@ public static IApplicationBuilder UseCookieAuthentication(this IApplicationBuild
                 throw new ArgumentNullException(nameof(app));
             }
 
-            return app.UseCookieAuthentication(options => { });
-        }
-
-        /// <summary>
-        /// Adds the <see cref="CookieAuthenticationMiddleware"/> middleware to the specified <see cref="IApplicationBuilder"/>, which enables cookie authentication capabilities.
-        /// </summary>
-        /// <param name="app">The <see cref="IApplicationBuilder"/> to add the middleware to.</param>
-        /// <param name="configureOptions">An action delegate to configure the provided <see cref="CookieAuthenticationOptions"/>.</param>
-        /// <returns>A reference to this instance after the operation has completed.</returns>
-        public static IApplicationBuilder UseCookieAuthentication(this IApplicationBuilder app, Action<CookieAuthenticationOptions> configureOptions)
-        {
-            if (app == null)
-            {
-                throw new ArgumentNullException(nameof(app));
-            }
-            if (configureOptions == null)
-            {
-                throw new ArgumentNullException(nameof(configureOptions));
-            }
-
-            var options = new CookieAuthenticationOptions();
-            configureOptions(options);
-
-            return app.UseMiddleware<CookieAuthenticationMiddleware>(options);
+            return app.UseCookieAuthentication();
         }
 
```

Comment:
All the other middleware also need this parameter-less extension to pull options from DI.